packages/offix-datastore/src/replication/api/MetadataModels.ts

```diff
@@ -0,0 +1,30 @@
+import { ModelSchema } from "../../ModelSchema";
+import { MODEL_METADATA, MODEL_METADATA_KEY, MUTATION_QUEUE, MUTATION_QUEUE_KEY } from "./Replicator";
+
+export const metadataModel = new ModelSchema<any>({
```

Comment:
Love it! Strongly typed metadata. It will make sense to merge this as.. 

@Eunovo work will invalidate this changes - we need to change id etc.

Reply:
@wtrocki yeah this was needed because we need to know the field names for all the stores upfront so we could create those, so I needed the full schema for the models and it just made sense to convert these to model schema too